File: 1876a_hinl.cpp
/**
 * @file 1876a_hinl.cpp
 * @author Ukeme Edet (ukemeedet2207@gmail.com)
 * @brief The solution to the problem Helmets in Night Light(https://codeforces.com/problemset/problem/1876/A)
 * @version 0.1
 * @date 2024-03-17
 *
 * @copyright Copyright (c) 2024
 *
 */

#include <bits/stdc++.h>

using namespace std;

/**
 * @brief The main function
 *
 * @return int
 */
int main(void)
{
	ios::sync_with_stdio(0);
	cin.tie(0);
	short t;
	int n, p, i;
	long long cost;
	vector<pair<int, int>> kc;

	cin >> t;
	while (t--)
	{
		cin >> n >> p;
		kc.resize(n);
		cost = p;
		n--;
		for (pair<int, int> &kci : kc)
			cin >> kci.second;
		for (pair<int, int> &kci : kc)
			cin >> kci.first;
		sort(kc.begin(), kc.end());
		i = 0;
		while (n && kc[i].first < p)
		{
			cost += (long long)kc[i].first * min(kc[i].second, n);
			n -= min(kc[i].second, n);
			i++;
		}
		if (n)
			cost += (long long)p * n;
		cout << cost << '\n';
	}
	return (EXIT_SUCCESS);
}

File: 1876a_hinl.py
"""
@file: 1876a_hinl.py
@author: Ukeme Edet (ukemeedet2207@gmail.com)
@brief: The solution to the problem Helmets in Night Light(https://codeforces.com/problemset/problem/1876/A)
@version: 1.0
@date: 2024-10-08

@copyright: Copyright (c) 2024
"""

from sys import stdin, stdout


def input():
    return stdin.readline().strip()


def print(string):
    return stdout.write(str(string) + "\n")


def main():
    t = int(input())
    for _ in range(t):
        n, p = [int(_) for _ in input().split()]
        ab = [
            [int(x), int(y)] for x, y in zip(input().split(), input().split())
        ]
        ab.sort(key=lambda x: x[1])
        cost, nr, i = p, 1, 0
        while nr < n:
            if ab[i][1] < p:
                cost += min(n - nr, ab[i][0]) * ab[i][1]
                nr += min(n - nr, ab[i][0])
                i += 1
            else:
                cost += p * (n - nr)
                break
        print(cost)


if __name__ == "__main__":
    main()
